### PR TITLE
PurgePugna section improvements

### PR DESCRIPTION
**Pull request by Shushishtok**
Fixes https://github.com/ModDota/dota-tutorial/issues/282

- Pugna now begins in the middle of the Radiant base, pacifist, with camera locked on him.
- Pugna then moves closer towards the player, then gets and uses a Blink Dagger to teleport inside the fountain.
- Works without and with skipping the dialog (properly waits for Pugna to arrive).
- The camera pans towards Pugna after the blinking.
- Pugna begins his monologue and is no longer pacifist and can be killed with Dragon Tail.
- Changed the hero to no longer automatically acquire targets, meaning he no longer automatically attacks Pugna.
- Reduced the wait time between each time Pugna is interrupted.
- Changed Pugna to die immediately when stunned with Dragon Tail, while saying his death line.
- Other very slight improvements.

### CODE REVIEW

File: game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts [hunk 1]
```diff
@@ -64,7 +95,7 @@ const start = (complete: () => void) => {
                     tg.loop(ctx => unitIsValidAndAlive(ctx[CustomNpcKeys.PurgePugna]), tg.audioDialog(LocalizationKey.Script_1_Leveling_4_2, LocalizationKey.Script_1_Leveling_4_2, ctx => ctx[CustomNpcKeys.PurgePugna])),
                     // Stop long dialog when we get attacked but not stunned. Let purge speak for at least a few seconds too.
                     tg.seq([
-                        tg.wait(2),
+                        tg.wait(0.1),
```
Comment (RobinKa) [inline game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts:98]: 2s was intentional here, so he starts talking for a bit instead of spamming "please dont interrupt me"

File: game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts [hunk 2]
```diff
@@ -43,18 +46,46 @@ const start = (complete: () => void) => {
 
         // Spawn purge and freeze player during initial dialog
         tg.immediate(_ => freezePlayerHero(true)),
-        tg.audioDialog(LocalizationKey.Script_1_Leveling_3, LocalizationKey.Script_1_Leveling_3, ctx => ctx[CustomNpcKeys.SlacksMudGolem]), // here comes pugna
-
-        tg.spawnUnit(CustomNpcKeys.PurgePugna, pugnaLocation, DotaTeam.BADGUYS, CustomNpcKeys.PurgePugna, true),
-        tg.immediate(ctx => getOrError(ctx[CustomNpcKeys.PurgePugna] as CDOTA_BaseNPC | undefined).SetAttackCapability(UnitAttackCapability.NO_ATTACK)),
-
+        tg.fork([
+            tg.seq([
```
Comment (RobinKa) [inline game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts:50]: Unnecessary seq

File: game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts [hunk 3]
```diff
@@ -43,18 +46,46 @@ const start = (complete: () => void) => {
...
+                tg.audioDialog(LocalizationKey.Script_1_Leveling_3, LocalizationKey.Script_1_Leveling_3, ctx => ctx[CustomNpcKeys.SlacksMudGolem]), // here comes pugna
+            ]),
+            tg.seq([
+                tg.spawnUnit(CustomNpcKeys.PurgePugna, pugnaLocation, DotaTeam.BADGUYS, CustomNpcKeys.PurgePugna, true),
+                tg.setCameraTarget(context => context[CustomNpcKeys.PurgePugna]),
+                tg.wait(FrameTime()),
```
Comment (RobinKa) [inline game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts:56]: Why is this needed?

Comment (Shushishtok) [reply game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts:56]: Had to add this because the unit spawns and is ordered to move in the same frame, which causes it to not yet be "initialized" enough to get commands. This caused it to ignore the move command.

File: game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts (hunk 2 above)
Comment (Shushishtok) [reply game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts:50]: Correct, removed.

File: game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts (hunk 1 above)
Comment (Shushishtok) [reply game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts:98]: This was actually an issue when you were immediately auto attacking him when he spawned, but that was resolved. Now he says "please don't interrupt me" once every 2 hits, which is about the time it takes for him to say that line.

Works fine on my end.

Comment (Shushishtok) [reply game/scripts/vscripts/Sections/Chapter1/SectionLeveling.ts]: REVERTED
